package/kedro_viz/models/experiment_tracking.py

```diff
@@ -99,3 +113,19 @@ def load_tracking_data(self, run_id: str):
 
 def get_dataset_type(dataset: AbstractVersionedDataSet) -> str:
     return f"{dataset.__class__.__module__}.{dataset.__class__.__qualname__}"
+
+
+def get_filename(dataset: AbstractVersionedDataSet) -> str:
+     dataset_description = dataset._describe()
+     return (dataset_description.get("filepath") or dataset_description.get("path")
+            ).name
+
+
+def matplotlib_writer_load(dataset: AbstractVersionedDataSet) -> str:
+    load_path = get_filepath_str(dataset._get_load_path(), dataset._protocol)
+    with open(load_path, "rb") as img_file:
```

Comment:
This might not work right on protocols outside local (like s3). We should do this instead (also in the metadata side panel):

```suggestion
    with dataset._fs.open(load_path, "rb", **dataset._fs_open_args_load) as img_file:
```

Reply:
this doesn't work. it says dataset does not have any function ._fs.open

Reply:
Hmm, I think this should work. Maybe you need `mode="rb"`?

Reply:
Done in [04165de](https://github.com/kedro-org/kedro-viz/pull/984/commits/04165de16f0857c066315ad5d5c76824f0ba96d0).